apps/android/app/src/main/java/chat/simplex/app/SimplexApp.kt

```diff
@@ -86,7 +86,16 @@ class SimplexApp: Application(), LifecycleEventObserver {
     Log.d(TAG, "onStateChanged: $event")
     withApi {
       when (event) {
-        Lifecycle.Event.ON_RESUME -> {
+        Lifecycle.Event.ON_RESUME -> CoroutineScope(Dispatchers.Default).launch {
+          /**
+           * When the app starts [chatModel.onboardingStage] is null, because [Lifecycle.Event.ON_RESUME] called sooner than chat is
+           * able to start. This is why [withTimeoutOrNull] waits for the moment when the value is not null and runs required steps afterward
+           * */
+          withTimeoutOrNull(10_000) {
+            while (isActive && chatModel.onboardingStage.value == null) {
+              delay(1000)
+            }
+          }
```

Comment:
this seems a hack - isn't there a better way? like some event sent from the place where chatModel.onboardingStage.value is set?